src/App.tsx

```diff
@@ -1,26 +1,54 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useContext } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getTodos } from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { DispatchContext } from './store/todoReducer';
+import { Action } from './types/actions';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [coverShow, setCoverShow] = useState<number[]>([]);
+
+  const dispatch = useContext(DispatchContext);
+
+  useEffect(() => {
+    getTodos()
```

Comment:
[Not critical]
The perfect way to handle this would be defining the whole `getTodos().then(...).catch(...)` logic as a separate function, maybe hook. `() => setErrorMessage('Unable to load todos'))` might be passed as an argument to it.

Reply:
Posting it here as well.
It can be a hook (preferrable):
```
const useGetInitialData = ({onSuccess, onError, onFinally}: {onSuccess?: (response) => void; onError?: (error) => void}, onFinally?: () => void) => {
    useEffect(() => {
        getTodos().then(onSuccess).catch(onError).finally(onFinally);
    }, []);
};

const App = () => {
    useGetInitialData({
    onSuccess: response => {
        dispatch({ type: Action.initialTodo, payload: response });
    },
    onError: () => setErrorMessage('Unable to load todos')
});
    ...
};
```
...or just a function:
```
const getInitialData = ({onSuccess, onError, onFinally}: {onSuccess?: (response) => void; onError?: (error) => void}, onFinally?: () => void) => {
    return getTodos().then(onSuccess).catch(onError).finally(onFinally);
};

const App = () => {
    useEffect(() => {
        useGetInitialData({
            onSuccess: response => {
                dispatch({ type: Action.initialTodo, payload: response });
            },
            onError: () => setErrorMessage('Unable to load todos')
        });
    }, []);
    ...
};
```